crates/solvers/src/domain/order.rs

```diff
@@ -0,0 +1,54 @@
+//! The domain object representing a CoW Protocol order.
+
+use crate::domain::eth;
+
+/// A CoW Protocol order in the auction.
+#[derive(Debug, Clone)]
+pub struct Order {
+    pub uid: Uid,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub side: Side,
+    pub class: Class,
+}
+
+/// UID of an order.
+#[derive(Debug, Clone, Copy)]
+pub struct Uid(pub [u8; 56]);
+
+/// The trading side of an order.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Side {
+    /// An order with a fixed buy amount and maximum sell amount.
+    Buy,
+    /// An order with a fixed sell amount and a minimum buy amount.
+    Sell,
+}
+
+/// The order classification.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Class {
+    Market,
+    Limit,
+    Liquidity,
+}
+
+/// A user order marker type.
+#[derive(Debug)]
+pub struct UserOrder<'a>(&'a Order);
```

Comment:
Is the deciding factor here that the order comes from a user or that it's surplus capturing? Because some users (in the future maybe all [[0](https://cowservices.slack.com/archives/C035N0YEB6J/p1673017823742939)]) can create liquidity orders.

Reply:
You are right. `UserOrder` is not a good name... Basically I need a name for a non-liquidity order (which we have called "user order" in the past, where I got the name from).

Not sure what it should be though...

Reply:
SurplusCapturingOrder is a mouthful but I guess it would be correct.

Reply:
We want to add [surplus capturing liquidity orders](https://www.notion.so/cownation/Surplus-Capturing-Liquidity-Orders-Advanced-Wait-for-CoW-006c00cfe9b3487a8cb6254b08859fc3) which would not be a "user order", since they shouldn't ever trade against on-chain liquidity, and therefore should not be valid order input to the Baseline solver.

Reply:
I will rename this to `NonLiquidityOrder` so it is consistent with the `usize`/`NonZeroUsize` analogy.

Reply:
ddbfdfea740c780fcc1223c0d03175862e25895a